doc/compatibility_matrix.rst

```diff
@@ -0,0 +1,15 @@
+.. _compatibility_matrix:
+
+Compatibility matrix
+====================
+
+Before you start working with the Sidewalk environment, familiarize yourself with the outlined version dependencies.
+The following table demonstrates the compatibility between the versions of the nRF Connect SDK and Sidewalk MCU SDK:
+

```

Comment:
minor: maybe add an information it's for nrf52840 chip?

Reply:
added to the line above